### PR TITLE
Remove DSA key type.

### PR DESCRIPTION
**Pull request by adiroiban**
Problem
======

This started due to the fact that key type is not listed for unknown errors.
While working on this I found out that our current Windows and OSX versions don't support DSA key types.

Due to this we should remove the key type from CSR generations.

DSA will be removed from TLS 1.3 and I don't think there are many people using it.

Changes
======

Removed --key-type

I have also comitted first code to allow running test on Win/OS X in our test environment.

How to test
========

reviewers: @alibotean 

check that changes make sense.

make test might still not work on your computer... it works on buildbot windows slave. .. i would like to continue working on this in #17

You can force the creating of the virtual env using .. it should create the same env as on buildslave
```
make ci_deps
```

You might need to clean your previous env with `make clean`

Please also try to use the demo command to generate certificates in unicode
```
python keycert-demo.py ssl-gen-key
```

### CODE REVIEW

**Comment by adiroiban** [comment]
Coveralls is stupid... as the percentage decreased due to fewer lines of code.

**Comment by adiroiban** [comment]
please take a look

thanks

needs-review

**Comment by alibotean** [comment]
Cleaning my env works, but when trying to run deps it fails:

```
$ make clean

$ make deps
New python executable in build/venv\Scripts\python.exe
Installing setuptools, pip...done.
make: build/venv/lib/Scripts/pip: Command not found
make: *** [local_env] Error 127
```

The file$ make clean                                             
                                                         
Adrian@Cyclops ~/chevah/chevah-keycert (19-show-ssl-type)
$ make deps                                              
New python executable in build/venv\Scripts\python.exe   
Installing setuptools, pip...done.                       
make: build/venv/lib/Scripts/pip: Command not found      
make: *** [local_env] Error 127                          
```

`pip.exe` is in `build/venv/Scripts/` instead of `build/venv/lib/Scripts/`

```
$ ls ./build/venv/Scripts/pip*                                                              
./build/venv/Scripts/pip.exe  ./build/venv/Scripts/pip2.7.exe  ./build/venv/Scripts/pip2.exe
```

**Comment by alibotean** [comment]
If I fix the path in the `Makefile` I get the following error: `Makefile:21: *** missing separator.  Stop.`

**Comment by alibotean** [comment]
Ran the make steps manually and it worked up the a point. It seems to get back wo the `pyCrypto` issue for missing MS Python compiler:

```
building 'Crypto.Random.OSRNG.winrandom' extension
warning: GMP or MPIR library not found; Not building Crypto.PublicKey._fastmath.
error: Unable to find vcvarsall.bat
```

File: Makefile [hunk 1]
```diff
@@ -2,22 +2,30 @@
 # Makefile for Chevah KeyCert project.
 #
 EXTRA_PYPI_INDEX='http://chevah.com/pypi/simple'
-UNAME=`uname`
-ifeq "$(UNAME)" "mingw"
-       BASE_PATH='build/venv/Scripts/'
+
+ifeq "$(MSYSTEM)" "MINGW32"
+       BASE_PATH='build/venv/lib/Scripts'
```
Comment (alibotean) [inline Makefile]: Should be `build/venv/Scripts`

File: release-notes.rst [hunk 2]
```diff
@@ -2,6 +2,15 @@ Relese notes for Chevah KeyCert
 ###############################
 
 
+1.3.2 - 14/04/2015
+==================
+
+* Update error message to show name for unknown key type.
```
Comment (alibotean) [inline release-notes.rst]: I don't think this is relevant any more.

**Comment by alibotean** [comment]
changes-approved

File: release-notes.rst (hunk 2 above)
Comment (adiroiban) [inline release-notes.rst]: true. thanks

File: Makefile (hunk 1 above)
Comment (adiroiban) [inline Makefile]: true

**Pull request closed without merging**